src/Microsoft.ApiDesignGuidelines.Analyzers/Core/IdentifiersShouldNotContainTypeNames.cs

```diff
@@ -0,0 +1,139 @@
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.IdentifiersShouldNotContainTypeNamesDescription), MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+
+        internal static ImmutableHashSet<string> s_types = ImmutableHashSet.CreateRange<string>(StringComparer.OrdinalIgnoreCase,
+            new []{
+                "char",
+                "wchar",
+                "int8",
+                "uint8",
+                "short",
+                "ushort",
+                "int",
+                "uint",
+                "integer",
+                "uinteger",
+                "long",
+                "ulong",
+                "unsigned",
+                "signed",
+                "float",
+                "float32",
+                "float64",
+                "int16",
+                "int32",
+                "int64",
+                "uint16",
+                "uint32",
+                "uint64",
+                "intptr",
+                "uintptr",
+                "ptr",
+                "uptr",
+                "pointer",
+                "upointer",
+                "single",
+                "double",
+                "decimal",
+                "guid",
+                "object",
+                "obj",
+                "string"
+            });
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessage,
+                                                                             DiagnosticCategory.Design,
+                                                                             DiagnosticSeverity.Warning,
+                                                                             isEnabledByDefault: true,
+                                                                             description: s_localizableDescription,
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/bb531486.aspx",
+                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.RegisterSymbolAction(AnalyzeNamedType, SymbolKind.NamedType);
```

Comment:
You can register multiple symbol kinds to one callback. i.e you can simply just have one AnalyzeSymbol that does what it does today, plus an added check for parameters for methods\properties. That would reduce duplicate code.